### PR TITLE
Smart album - On This Day - new feature

### LINKED ISSUE #1599: [Enhancement] Smart album "On This Day"
I have a large collection of photos and I needed functionality to see all the photos taken on this day. This way, it's easier to choose my favorite photos and it's interesting to see what's been photographed each year on this day.
If you like this functionality I can create pull requests.

### PR DESCRIPTION
**Pull request by aldjordje**
The smart album "On This Day" will allow users to see all the photos taken on this day.
Closes https://github.com/LycheeOrg/Lychee/issues/1599

### CODE REVIEW

File: app/Locale/French.php
Comment (ildyria) [inline app/Locale/French.php]:
```suggestion
			'ON_THIS_DAY' => 'Ce jour-là',
```

**Comment by ildyria** [comment]
@aldjordje I sent you an Triage invite so you are able to run the pipelines without waiting for our approval. :)

File: app/Locale/Dutch.php
Comment (ildyria) [inline app/Locale/Dutch.php]:
```suggestion
			'ON_THIS_DAY' => 'Deze Dag',
```
(from my basic knowledge of Dutch :') )
@nagmat84 should be able to give you the one in German

File: app/Locale/German.php
Comment (qwerty287) [inline app/Locale/German.php]:
```suggestion
			'ON_THIS_DAY' => 'An diesem Tag',
```

File: app/SmartAlbums/OnThisDayAlbum.php
Comment (qwerty287) [inline app/SmartAlbums/OnThisDayAlbum.php]: Is it intentional to not use check the year here?

Comment (aldjordje) [reply app/SmartAlbums/OnThisDayAlbum.php]: Yes, this will return all the photos taken on this day from every year.

**Comment by ildyria** [comment]
@aldjordje is this ready for review ? :)

**Comment by aldjordje** [comment]
@ildyria Thank you for enabling exceptions for 500 errors to be displayed. Without it, I couldn't figure out what the problem was :)

**Review by ildyria — CHANGES_REQUESTED** [review]
99% LGTM.

Please add the following standalone test to `AlbumTest.php`.

1. Get today date
2. Log as admin
3. Upload photo -> get the photo ID
4. Get photo model and set taken_at of the picture to today - 1 year.
5. Open "On this day"
6. Check that photo ID is present in the response.
7. Log out.

This will guarantee that the SQL request is also working. :)

Some pictures do not have a `taken_at` (exif data missing), maybe in those case we could consider the `created_at` instead ? See https://github.com/aldjordje/Lychee/pull/1/files :)

**Comment by aldjordje** [comment]
@ildyria I was thinking about the test and I couldn't  think of a way to write it. I'll add it today.

**Review by ildyria — APPROVED** [review]
LGTM.

**Pull request merged**
